### DISCUSSION TITLE
Why ebiten.FPSModeVsyncOffMinimum is deprecated ?

### DISCUSSION THREAD

**Discussion by rverpillot** [hajimehoshi/ebiten#3107] [Q&A]
I'm coding a desktop application (not a game) with Ebiten, and because it's deprecated I replaced `ebiten.SetFPSMode(ebiten.FPSModeVsyncOffMinimum)` by `ebiten.SetScreenClearedEveryFrame(false)` and redraw only when necessary. But the CPU usage has increased significantly (CPU is between 5% and 15% whereas previously it was 1% or lower).

The big difference is that by using `FPSModeVsyncOffMinimum` ebiten engine uses `glfw.WaitEvents()` instead of `glfw.PollEvents()`and the CPU usage is significantly lower. 
I tried to reduce the TPS, but it doesn't work.

If `ebiten.FPSModeVsyncOffMinimum` is deprecated, is it possible to use `glfw.WaitEvents()` in another way in ebiten ?

**Comment by hajimehoshi** [comment]
https://github.com/hajimehoshi/ebiten/issues/2338
https://github.com/hajimehoshi/ebiten/issues/2342

As far as I remember the discussion, the reason is to simplify the model of Ebitengine, especially Update.

**Comment by rverpillot** [comment]
Please, do not remove this feature, because there is not a true alternative (too many call to `PollEvents` in `Update`).

**Reply by tinne26** [reply]
While there's not a true alternative to it, `FPSModeVsyncOffMinimum` wasn't a perfect solution by itself either. In particular, it had the possibly surprising/problematic behavior of easily raising your TPS/FPS to many hundreds per second when you move your mouse around (especially if you do it fast), which can cause a whole new set of issues.

A good "minimum" model would require async input and decoupling of TPS and FPS from input. In fact, it would also probably require manual buffer swapping and more full control over the main loop. This is too different from the current model, and it doesn't mix well.

All in all, there are many reasons why `FPSModeVsyncOffMinimum`, at least at the moment, doesn't have a natural place to fit in / be properly implemented in Ebitengine.
